### DISCUSSION TITLE
ctags "No such file or directory" error reported in index logs

### DISCUSSION THREAD

**Discussion by chenchuanliang** [oracle/opengrok#4121] [General]
![EX)N3SI%42F{{26 F%_9UX](https://user-images.githubusercontent.com/20430797/206599630-e103da59-4eaf-46e0-9bde-21587eab060a.png)
![38fa096a2e3078b146ad6c4124b1844](https://user-images.githubusercontent.com/20430797/206599654-626ec810-61e7-4538-803f-1d8d90393d89.jpg)
Our project has reported many such errors when indexing, but the actual file exists, and the search is normal after indexing. I'm worried about these errors.
![H@R(1W650M~7 AT~$319ERE](https://user-images.githubusercontent.com/20430797/206600069-d0578e52-55e4-4782-83aa-2c62a1d839ac.png)

Why does ctags report this error?
Merry Christmas in advance!

**Comment by vladak** [comment]
What is special about these files ? How do you run the indexer ?

**Reply by chenchuanliang** [reply]
> What is special about these files ? How do you run the indexer ?

There is nothing special about these files.
CMD:
./opengrok-indexer -J=-Xmx32g -J=-server -a ~/opengrok/dist/lib/opengrok.jar -- -c /usr/local/bin/ctags -m 32 -s /workspace1 -d /opengrokData --progress -Q on --remote off -q -P -S -G -W ~/opengrok/etc/configuration.xml -U http://localhost:8080/source
Thank you

**Reply by vladak** [reply]
I meant: does this error happen for all the files being indexed or just for some of them ? Where does the `/usr/local/bin/ctags` come from ? How it was installed ?

**Reply by chenchuanliang** [reply]
> I meant: does this error happen for all the files being indexed or just for some of them ? Where does the `/usr/local/bin/ctags` come from ? How it was installed ?

Yes,it happened for all the files being indexed./usr/local/bin/ctags,this is the ctags install path.I use the ubuntu 18.04 and install ctags by "snap install universal-ctags" cmd.Then make a link to `/usr/local/bin/ctags`.
Thank you

**Reply by chenchuanliang** [reply]
> > I meant: does this error happen for all the files being indexed or just for some of them ? Where does the `/usr/local/bin/ctags` come from ? How it was installed ?
> 
> Yes,it happened for all the files being indexed./usr/local/bin/ctags,this is the ctags install path.I use the ubuntu 18.04 and install ctags by "snap install universal-ctags" cmd.Then make a link to `/usr/local/bin/ctags`. Thank you

I use 1.7.39 index version

**Reply by chenchuanliang** [reply]
> I meant: does this error happen for all the files being indexed or just for some of them ? Where does the `/usr/local/bin/ctags` come from ? How it was installed ?

![SEM$ZVR15_ 4@P}L7)X7H88](https://user-images.githubusercontent.com/20430797/207014305-6ece9d9e-d731-4713-aead-a25337e29dbb.png)
ctags version

**Reply by vladak** [reply]
This does not answer any of my questions.

**Reply by chenchuanliang** [reply]
> I meant: does this error happen for all the files being indexed or just for some of them ? Where does the `/usr/local/bin/ctags` come from ? How it was installed ?

In addition, there is a problem that the definition after indexing cannot be searched. I am not sure where it is caused. Recently, the search server was changed.

**Reply by chenchuanliang** [reply]
> This does not answer any of my questions.

Hi,
Yes,it happened for all the files being indexed./usr/local/bin/ctags,this is the ctags install path.I use the ubuntu 18.04 and install ctags by "snap install universal-ctags" cmd.Then make a link to /usr/local/bin/ctags.
Is this answer?
Thanks

**Reply by vladak** [reply]
That's the problem - snap based ctags installation is toxic for OpenGrok as it is constrained in what paths it is allowed to access. See #3342

Also, this is also mentioned on https://github.com/oracle/opengrok/wiki/How-to-setup-OpenGrok

Unfortunately, `ctags --version` does not give any indication that it is snap based, otherwise I'd add a check.

**Reply by chenchuanliang** [reply]
> That's the problem - snap based ctags installation is toxic for OpenGrok as it is constrained in what paths it is allowed to access. See #3342
> 
> Also, this is also mentioned on https://github.com/oracle/opengrok/wiki/How-to-setup-OpenGrok
> 
> Unfortunately, `ctags --version` does not give any indication that it is snap based, otherwise I'd add a check.

Thank you!
I will reinstall ctags and recheck
